### ISSUE TITLE
[Suggestion] Add `DeathLog` mod support

### ISSUE THREAD

**Issue by latipun7** [TheCSMods/mc-better-stats#108]
### What is this suggestion for?

Mod Itself

### Suggestion

[`DeathLog`](https://modrinth.com/mod/deathlog) has a button to access death log on vanilla _Statistics_ screen. Please add support to add `DeathLog` button on BSS. 
Thank you.

### Code of Conduct

- [X] I agree that I will be available later for any follow-up questions to help with implementing the suggested thing.

**Comment by TheCSDev** [comment]
Hello, and thank you for the suggestion!  
Aside from thos issue being a duplicate of #13, there are unfortunately technical limitations preventing me from implementing this feature.

In order for me to implement this, the other mod (in this case `DeathLog`), would have to provide some sort of API or something that functions like an API, that is reliable, and that other mod developers can use in order to open the `DeathLog` `Screen` from another mod such as this one.  
Now, as you can probably already tell by now, `DeathLog` features no APIs. This basically means I do not have any tools to actually make this possible.

I did however, eventually make my own APIs for this mod, back in `v3.0`. As of right now, `betterstats` has APIs for various things, including:
- Creating custom statistics tabs, and
- Adding custom items to the menu-bar (likely what `DeathLog` wants)

And with the API I implemented into this mod, I am hoping other mod developers can do stuff to make their mods more available through this mod. I of-course, am trying to find ways (especially as of lately) to support other mods and their modded features, but in some cases such as this one, I simply don't have the tools to make it possible.

In other words, I am basically saying that perhaps the developer of `DeathLog` could either implement an API in their own mod, or use this mod's API to implement support themselves. If they or any other mod developer out there, ever needs help using my APIs, they may feel free to reach out to me and ask me anything.

So hopefully that explains things for you. If you need anything else or have any questions, feel free to let me know. As for this issue, I think it will have to be closed as "Not planned".  
Have a great day (or night)!

**Comment by latipun7** [comment]
Thank you for the reply, I understand. 
Sorry, I forgot to search the issue before opening new issue 🙏

**Comment by TheCSDev** [comment]
It's been quite a while since I posted https://github.com/gliscowo/deathlog/issues/38  
Given there is no response for now, I have decided to close this issue so it doesn't end up open indefinitely. If a way to implement this presents itself, then this issue may be reopened and the feature may be implemented.  
Thanks again, and have a great day!

**Comment by EGOIST1372** [comment]
@TheCSDev Hey there
As of the deathlog dev answered 19 days after your last response in this issue here
I just wanted to know is there any plan to see an implementation of it using those info (I didn't want to open a duplicate issue)
Thanks in advance <3

**Comment by TheCSDev** [comment]
Hello and thanks for reminding about DeathLog!

I am currently building a rewrite of all my mods (that's gonna take a good while) as a means of dealing with the mod-breaking changes the game's updates did to my mods.
Once I am done with the rewrites (which will likely release as v5.0.0), I will look into DeathLog and see if I can implement that as well.
As for v3.X and v4.X, I have no plans to further maintain it, aside from porting it to future game versions.

Will re-open this issue to increase the odds of me remembering to look into it someday in the future when my rewrites are done.
Thanks for reaching out and have a great week!
